### PR TITLE
Fix EZP-23953: Legacy session events not triggered any more

### PR DESCRIPTION
**Pull request by lolautruche**
> https://jira.ez.no/browse/EZP-23953
> https://jira.ez.no/browse/EZP-23904

As of 5.3, authentication is completely managed through Symfony. One side effect is that legacy session events are not triggered any more (either via `ezpEvent` or `eZSession::triggerCallback()`), and thus listeners won't be executed. This is because it is now entirely up to Symfony to deal with the session, including session migration after authentication.
**Concrete illustration of this regression :** When using the legacy shop module, items in the shopping basket are lost after logging in.
In short, when the session is migrated after the user has been authenticated, the basket table, storing each entry per session ID, won't be updated at that time, resulting the loss of the basket for current user.

This PR introduces proxies for session storage and session save handler. They use legacy callbacks to trigger the appropriate events, just like what is done in legacy session handlers.

### CODE REVIEW

**Comment by lolautruche** [comment]
Review ping @andrerom @bdunogier @yannickroger @pspanja @glye

**Comment by yannickroger** [comment]
Before reading the code, just want to say thank you for the description :)

File: mvc/Session/LegacySessionProxy.php [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    public function open( $savePath, $sessionName )
+    {
+        $return = true;
+        if ( $this->sessionHandler )
+        {
+            $return = (bool)$this->sessionHandler->open( $savePath, $sessionName );
+        }
+
+        if ($return === true)
```
Comment (andrerom) [inline mvc/Session/LegacySessionProxy.php]: cs

**Comment by andrerom** [comment]
+1, looks great and thanks for the write up indeed :)

**Comment by yannickroger** [comment]
+1

**Comment by lolautruche** [comment]
This one being sensitive, I'll wait at least to have 3 *+1*

**Comment by glye** [comment]
YESSSH!!!
The first commit fixes EZP-23904 for me.
Very +1.

**Comment by lolautruche** [comment]
OK, just fixed the tests.

**Comment by yannickroger** [comment]
@glye That is a great news !
